src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1511,6 +1515,18 @@ private boolean isRelevantBuildData(BuildData bd) {
         return false;
     }
 
+    public String getRef(PreBuildMergeOptions options) {
+        final String mergeTarget = options.getMergeTarget();
+        if (StringUtils.isNotBlank(mergeTarget)) {
+            return mergeTarget;
+        } else if (getBranches().size() == 1) {
+            return getBranches().get(0).getName();
+        } else {
+            throw new NotImplementedException(
```

Comment:
Please avoid runtime exceptions, IMO the best way would be to...
* return null
* add <code>@CheckForNull</code> annotation
* handle issue and throw <code>AbortException</code> in perform()